HypothesisPlugin.inc.php

```diff
@@ -76,6 +84,83 @@ function changePdfjsPath($output, $templateMgr) {
 		return $newOutput;
 	}
 
+	/**
+	 * Adds Hypothesis tab configuration so sidebar opens automatically when PDF has annotations
+	 * @param $output string
+	 * @param $templateMgr TemplateManager
+	 * @return $string
+	 */
+	public function addHypothesisConfig($output, $templateMgr) {
+		if (preg_match('/<div[^>]+id="pdfCanvasContainer/', $output, $matches, PREG_OFFSET_CAPTURE)) {
+            $posMatch = $matches[0][1];
+			$config = $templateMgr->fetch($this->getTemplateResource('hypothesisConfig.tpl'));
+
+            $output = substr_replace($output, $config, $posMatch, 0);
+            $templateMgr->unregisterFilter('output', array($this, 'addHypothesisConfig'));
+        }
+		return $output;
+	}
+
+	public function addAnnotationNumberViewers($hookName, $args) {
+		$templateMgr = $args[0];
+		$template = $args[1];
+		$pagesToInsert = [
+			'frontend/pages/indexServer.tpl',
+			'frontend/pages/preprint.tpl',
+			'frontend/pages/preprints.tpl',
+			'frontend/pages/sections.tpl',
+			'frontend/pages/indexJournal.tpl',
+			'frontend/pages/article.tpl',
+			'frontend/pages/issue.tpl'
+		];
+
+		if (in_array($template, $pagesToInsert)) {
+            $request = Application::get()->getRequest();
+
+            $jsUrl = $request->getBaseUrl() . '/' . $this->getPluginPath() . '/js/addAnnotationViewers.js';
+            $styleUrl = $request->getBaseUrl() . '/' . $this->getPluginPath() . '/styles/annotationViewer.css';
+
+            $templateMgr->addJavascript('AddAnnotationViewers', $jsUrl, ['contexts' => 'frontend']);
+            $templateMgr->addStyleSheet('AnnotationViewerStyleSheet', $styleUrl, ['contexts' => 'frontend']);
+        }
+
+		return false;
+	}
+
+	public function addAnnotationsHandler($hookName, $args) {
+		$page = $args[0];
+		if ($page == 'annotations') {
+			$this->import('pages.annotations.AnnotationsHandler');
+			define('HANDLER_CLASS', 'AnnotationsHandler');
+			return true;
+		}
+		return false;
+	}
+
+	public function setupHypothesisHandler($hookName, $args) {
+		$component = &$args[0];
+        if ($component == 'plugins.generic.hypothesis.controllers.HypothesisHandler') {
+            return true;
+        }
+        return false;
+	}
+
+	public function addTasksToCrontab($hookName, $args) {
+		$taskFilesPath = &$args[0];
+        $taskFilesPath[] = $this->getPluginPath() . DIRECTORY_SEPARATOR . 'scheduledTasks.xml';
+        return false;
+	}
+
+	public function addHandlerURLToJavaScript()
+    {
+        $request = Application::get()->getRequest();
+        $templateMgr = TemplateManager::getManager($request);
+        $handlerUrl = $request->getDispatcher()->url($request, ROUTE_COMPONENT, null, 'plugins.generic.hypothesis.controllers.HypothesisHandler');
```

Comment:
Are we sure that the scheduled task is executed successfully? I am getting a 

```
PHP Fatal error:  Uncaught Error: Call to a member function url() on null in ojs-3.3.0-17\plugins\generic\hypothesis\HypothesisPlugin.inc.php:158
```

When trying to execute `php runScheduledTasks.php -h`